### PR TITLE
add major app code base

### PR DESCRIPTION
**Pull request by Morenayd**

### CODE REVIEW

**Comment by Morenayd** [comment]
The change theme button works, but only after the third tap when the app is first launched. Working on fixing this soon.

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+class MainActivity : AppCompatActivity() {
+
+    companion object {
+        private const val CURRENT_QUOTE = "CURRENT_QUOTE"
+    }
+
+    private val quotes = mutableListOf("Keep moving forward", "Don't just fly, soar", "Today is a good day to try", "Believe you can, then you will")
+    private var quote = quotes.random()
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:24]: Make sure to add some spacing between functions & properties. Usually just 1 empty line is fine!

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        super.onCreate(savedInstanceState)
+        if (AppCompatDelegate.getDefaultNightMode() == AppCompatDelegate.MODE_NIGHT_YES) {
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:27]: I'd pull this out in a separate function called `applyDefaultDeviceTheme()` or something similar! But nice work!

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            setTheme(R.style.DarkTheme)
+        } else {
+            setTheme(R.style.LightTheme)
+        }
+        setContentView(R.layout.activity_main)
+        val quoteTextView = findViewById<TextView>(R.id.quote_textview)
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:33]: I'd use kotlin synthetics or ViewBinding instead. findViewById is clunky and a bit annoying to use!

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        details_textview.movementMethod = LinkMovementMethod.getInstance()
+        quoteTextView.text = if (savedInstanceState != null) {
+            savedInstanceState.getString(CURRENT_QUOTE) ?: quotes.random()
+        } else {
+            setQuote()
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:38]: I'd have this into a separate function named `setQuoteText(savedInstanceState)`! But nice work once more.

Also, I'd rename `setQuote()` to `getQuote()` as it generally gets a new quote, it doesn't set anything!

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        }
+        val handler = Handler()
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:40]: I'd pull this out in a new function.

Also I'd pull out the Handler in a property, and then cancel it/release it in `onStop` or something, and restart it in `onStart` or `onCreate`.

This way you make sure not to leak any memory! I'd also pull this entire piece of code into a separate function!

But nice work on the Handler, way to go!

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    private fun changeTheme() {
+        if (AppCompatDelegate.getDefaultNightMode() == AppCompatDelegate.MODE_NIGHT_YES) {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+        } else {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+        }
+    }
+
+    private fun share() {
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:78]: Maybe rename this to `shareInformation()`.

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        startActivity(getShareIntent())
+    }
+
+    private fun getShareIntent(): Intent {
+        val shareIntent = Intent(Intent.ACTION_SEND)
+        shareIntent.setType("text/plain").putExtra(Intent.EXTRA_TEXT, getString(R.string.my_details))
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:84]: It's also important to check if there is an app which can perform the ACTION_SEND.

You can check out some examples here: https://stackoverflow.com/questions/15407502/how-to-check-if-an-intent-can-be-handled-from-some-activity

File: Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        return shareIntent
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(CURRENT_QUOTE, quote)
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/java/com/example/meetme/MainActivity.kt:90]: Make sure to put the data before passing the outState to the parent!

File: Week3/MeetMe/app/src/main/res/layout-land/activity_main.xml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="match_parent"
+        android:gravity="center_horizontal|center_vertical"
+        android:orientation="vertical"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <ImageView
+            android:id="@+id/imageView"
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/res/layout-land/activity_main.xml:21]: I'd be careful about the naming here.

I'd use descriptive names such as: "userAvatar", "userBio", "userHobbies" etc. or "hobbiesText" and similar.

It's very hard to understand what's going on here.

File: Week3/MeetMe/app/src/main/res/layout-land/activity_main.xml [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        android:layout_marginLeft="@dimen/margin_large"
+        android:layout_marginEnd="@dimen/margin_large"
+        android:layout_marginRight="@dimen/margin_large"
+        android:layout_marginBottom="@dimen/margin_standard"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/linear_layout"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <LinearLayout
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/res/layout-land/activity_main.xml:67]: You can just use the TextView, or a FrameLayout here!

File: Week3/MeetMe/app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+        android:layout_marginTop="@dimen/margin_large"
+        android:layout_marginEnd="@dimen/margin_large"
+        android:layout_marginRight="@dimen/margin_large"
+        android:layout_marginBottom="@dimen/margin_standard"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/imageView">
+
+        <LinearLayout
```
Comment (filbabic) [inline Week3/MeetMe/app/src/main/res/layout/activity_main.xml:66]: Same here, no need for the LinearLayout!

Also, naming can be better as described above!

**Pull request merged**